.github/workflows/linter.yml

```diff
@@ -0,0 +1,18 @@
+name: golangci-lint
+on:
+    push:
+      branches: [ "main" ]
+    pull_request:
+
+permissions:
+  contents: read
+  pull-requests: read
+
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Run golangci-lint
+        run: docker run --rm -v $(pwd):/app -w /app/eng/_util golangci/golangci-lint:v2.0 golangci-lint run -v
```

Comment:
`actions/checkout` and the image are unpinned dependencies.

For the golangci-lint one, could we use a `go tool` go.mod dependency instead, so that I can be sure I'm running the same checks locally?

Reply:
updated PTAL